ratelimit/utils.py

```diff
@@ -157,7 +157,7 @@ def get_usage_count(request, group=None, fn=None, key=None, rate=None,
     cache_key = _make_cache_key(group, rate, value, method)
     time_left = _get_window(value, period) - int(time.time())
     initial_value = 1 if increment else 0
-    added = cache.add(cache_key, initial_value)
+    added = cache.add(cache_key, initial_value, period)
```

Comment:
Does using exactly `period`, instead of e.g. `period + 1` cause any risks with one-second windows?

Reply:
Probably there can be some edge cases where key is expired before it is fetched (because of network problem or something). 
Also keeping keys a few seconds more than actually needed will not hurt at all.